### PR TITLE
Fixed all the warnings on rPi linux builds

### PR DESCRIPTION
**Pull request by dacarley**

### CODE REVIEW

File: src/engines/synths/nuke/nuke.cpp [hunk 1]
```diff
@@ -95,6 +96,7 @@ namespace otto::engines {
       case ui::Rotary::Green: return knobs[1].rotation;
       case ui::Rotary::White: return knobs[2].rotation;
       case ui::Rotary::Red: return knobs[3].rotation;
+      default: throw util::exception("Unknown rotary value");
```
Comment (topisani) [inline src/engines/synths/nuke/nuke.cpp]: we should not add a default when all enum options are exhausted, especially not scoped enums

File: src/util/bytefile.hpp [hunk 2]
```diff
@@ -122,6 +122,8 @@ namespace otto::util {
           return "Exception thrown"; break;
         case Type::PastEnd:
           return "Past the end of the file"; break;
+	default:
```
Comment (topisani) [inline src/util/bytefile.hpp]: this one again - its very important that there is no default case, so we get the warning if we add a `Type` but forget to add it here

File: src/util/jsonfile.cpp [hunk 3]
```diff
@@ -56,6 +56,7 @@ namespace otto::util {
     case ErrorCode::missing_header: return "OTTO Preset header missing";
     case ErrorCode::unknown_version: return "Unknown preset format version";
     case ErrorCode::invalid_data: return "Invalid preset data. Take a look at the specification again";
+    default: return "Unknown ErrorCode";
```
Comment (topisani) [inline src/util/jsonfile.cpp]: here

File: src/engines/synths/nuke/nuke.cpp (hunk 1 above)
Comment (dacarley) [reply src/engines/synths/nuke/nuke.cpp]: Switched to a pragma to quiet the compiler.

File: src/util/bytefile.hpp (hunk 2 above)
Comment (dacarley) [reply src/util/bytefile.hpp]: Ditto

File: src/util/jsonfile.cpp (hunk 3 above)
Comment (dacarley) [reply src/util/jsonfile.cpp]: Ditto

File: src/engines/synths/nuke/nuke.cpp [hunk 4]
```diff
@@ -88,6 +87,8 @@ namespace otto::engines {
     return true;
   }
 
+#pragma GCC diagnostic push 
+#pragma GCC diagnostic ignored "-Wreturn-type"
```
Comment (topisani) [inline src/engines/synths/nuke/nuke.cpp]: I have no idea why this is the warning to disable for this, but lets roll with it

**Pull request merged**